pkg/cmd/server/apis/config/v1/types.go

```diff
@@ -542,6 +542,8 @@ type MasterNetworkConfig struct {
 	// For security reasons, you should ensure that this range does not overlap with the CIDRs reserved for external ips,
 	// nodes, pods, or services.
 	IngressIPNetworkCIDR string `json:"ingressIPNetworkCIDR"`
+	// VXLANPort is the VXLAN port used by the cluster
+	VXLANPort uint32 `json:"vxlanPort"`
```

Comment:
This is expected to be used only in case of openshift on vmware NSX, then why not make it optional param.
VXLANPort *uint32 `json:"vxlanPort,omitempty"`

Reply:
I think making it optional would only make sense if VXLAN was optional. If we allow it to be `nil` here, we just need to hardcode the default value somewhere else instead.

Reply:
the external api and the surrounding autogenerated code does not support uint16, Instead of converting internally I thought to leave it as uint32.